src/main/scala/cromwell/util/ConfigUtil.scala

```diff
@@ -0,0 +1,91 @@
+import org.slf4j.{LoggerFactory, Logger}
+import scala.reflect.{ClassTag, classTag}
+import scala.collection.JavaConversions._
+import scala.util.Try
+import scalaz._
+import Scalaz._
+
+object ConfigUtil {
+
+  val validationLogger = LoggerFactory.getLogger("ConfigurationValidation")
+  
+  class ConfigValidationException(context: String, validationException: ConfigException.ValidationFailed)
+    extends ConfigException.ValidationFailed(validationException.problems()) {
+    override def getMessage: String = {
+      val problems = validationException.problems().map(_.problem()).mkString(", ")
+      s"$context configuration validation failed : $problems"
+    }
+  }
+
+  implicit class EnhancedConfig(val config: Config) extends AnyVal {
+
+    def keys = config.entrySet().toSet map { v: java.util.Map.Entry[String, ConfigValue] => v.getKey }
+
+    def getStringOption(key: String): Option[String] = {
+      Try(config.getString(key)) match {
+        case scala.util.Success(value) => Option(value)
+        case scala.util.Failure(e: ConfigException.Missing) => None
+        case scala.util.Failure(e) => throw e
+      }
+    }
+
+    /**
+     * Checks that the configuration is valid compared to refConfig (i.e. at least all same keys with same type).
+     * Wrap the exception to add some context information about what this config is about.
+     */
+    def checkValidWithWarnings(refConfig: ReferenceConfiguration) = {
+      try {
+        config.checkValid(refConfig.requiredConfig)
+        warnNotRecognized(refConfig)
+      } catch {
+        case e: ConfigException.ValidationFailed => throw new ConfigValidationException(refConfig.context, e)
+        case t: Throwable => throw t
+      }
+    }
+
+    /**
+     * For keys that are in the configuration but not in the reference configuration, log a warning.
+     */
+    def warnNotRecognized(refConf: ReferenceConfiguration) = {
+      keys.diff(refConf.keys) match {
+        case warnings if warnings.nonEmpty => validationLogger.warn(s"Unrecognized configuration key(s) for ${refConf.context}: ${warnings.mkString(", ")}")
+        case _ =>
+      }
+    }
+
+    /**
+     * Validates that the value for this key is a well formed URL.
+     */
+    def validateURL(key: String): ValidationNel[String, URL] = key.validateAny { url =>
+      new URL(config.getString(url))
+    }
+
+  }
+
+  implicit class EnhancedValidation[I <: AnyRef](val value: I) extends AnyVal {
+    /**
+     * Validates this value by applying validationFunction to it and returning a Validation:
+     * Returns successNel upon success.
+     * If an exception is thrown and is a subtype of E, return failureNel with the exception message.
+     * @param validationFunction function that should throw an exception if this value is found not to be valid
+     * @tparam O return type of validationFunction
+     * @tparam E Restricts the subtype of Exception that should be caught during validation
```

Comment:
Hmm I don't know, I imagine in some cases you might not want to catch ALL kinds of `Exception`s here (I'd say only "validation-related" exceptions should be caught), which is why I added this, but it's true that in practice the validation functions are so simple currently that it's not worth the trouble to restrict the Exception scope. 
I personally think it does not hurt to have it since you can completely ignore it on the caller side if you don't want to bother with it, but I'm not against removing it entirely if it's confusing.